shared/middleware/helpers/bucket_helper.rb

```diff
@@ -316,31 +316,43 @@ def restore_previous_version(encrypted_channel_id, filename, version_id, user_id
 
     if version_restored
       # If we get this far, the restore request has succeeded.
-      FirehoseClient.instance.put_record(
-        study: 'project-data-integrity',
-        study_group: 'v2',
-        event: 'version-restored',
-
-        # Make it easy to limit our search to restores in the sources bucket for a certain project.
+      log_restored_file(
         project_id: encrypted_channel_id,
-        data_string: @bucket,
-
         user_id: user_id,
-        data_json: {
-          restoredVersionId: version_id,
-          newVersionId: response.version_id,
-          bucket: @bucket,
-          key: key,
-          filename: filename,
-        }.to_json
+        filename: filename,
+        source_version_id: version_id,
+        new_version_id: response.version_id
       )
     end
 
-    response
+    response.to_h
   end
 
   protected
 
+  def log_restored_file(project_id:, user_id:, filename:, source_version_id:, new_version_id:)
+    owner_id, channel_id = storage_decrypt_channel_id(project_id)
+    key = s3_path owner_id, channel_id, filename
+    FirehoseClient.instance.put_record(
+      study: 'project-data-integrity',
+      study_group: 'v2',
```

Comment:
Should we bump the study group for the `project-data-integrity` study with this change, or shortly thereafter?

Reply:
I say no, but I could be convinced otherwise - in the past it seems like we've only bumped to avoid confusion in situations where additional logging or changes to logging format could be misinterpreted as an increase/decrease in errors. Since these changes are meant to decrease errors, I think we want to not-bump the study group?

Reply:
Please go ahead and bump it to v3, here and 1 other place in this file, and in one place in project.js

Reply:
oops, I missed @epeach 's comment. I'm fine with not bumping. we can always use the date to filter if we need to, so the cost is pretty low if we do not bump but wish we did later.